lib/rules/padded-blocks.js

```diff
@@ -11,19 +11,44 @@
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
-    var requirePadding = context.options[0] !== "never";
+    var options = {};
+    var config = context.options[0] || "always";
+
+    if (typeof config === "string") {
+        options.blocks = config !== "never";
```

Comment:
Since the schema guarantees that options will always be `"always"` or `"never"`, `config === "always"` is functionally equivalent and IMO easier to comprehend. (Here and for the individual `blocks` and `switches` options below.)